src/app/api/services/initialization.py

```diff
@@ -0,0 +1,167 @@
+        params["geo_file_path"]
+    )
+
+    data["compute_parameters"]["restart_parameters"]["start_datetime"] = params[
+        "start_datetime"
+    ]
+    data["compute_parameters"]["restart_parameters"]["lite_channel_restart_file"] = (
+        restart_file.__str__()
+    )
+    data["compute_parameters"]["forcing_parameters"]["nts"] = params["nts"]
+    data["compute_parameters"]["forcing_parameters"]["qlat_input_folder"] = params[
+        "qlat_input_folder"
+    ]
+
+    data["output_parameters"]["stream_output"]["stream_output_directory"] = (
+        output_dir.__str__()
+    )
+
+    with open(tmp_yaml, "w") as file:
+        yaml.dump(data, file)
+
+    return tmp_yaml
+
+
+def create_params(
+    lid: str,
+    feature_id: str,
+    hy_id: str,
+    initial_start: float,
+    start_time: str,
+    num_forecast_days: int,
+    settings: Settings,
+) -> Dict[str, str]:
+    """Generating the parameters to be added to the T-Route config files
+
+    Parameters:
+    -----------
+    lid: str
+        The location ID for the RFC point
+    feature_id: str
+        The feature ID, or COMID/hf_id, from the RFC point
+    hy_id: str
+        The HYFeatures ID from the RFC point
+    initial_start: float
+        The initial starting flow (m3/s) for Routing
+    start_time: str
+        The starting time ("%Y-%m-%dT%H:%M:%S") for when routing begins
+    num_forecast_days: int
+        The number of days we are going to route
+    settings: Settings
+        The T-route BaseSettings
+
+    Returns:
+    --------
+    Dict[str, str]:
+        The parameters to be added to the t-route config file
+    """
+    dt = datetime.strptime(start_time, "%Y-%m-%dT%H:%M:%S")
+    start_datetime = dt.strftime("%Y-%m-%d_%H:%M")
+
+    nts = 288 * num_forecast_days
+
+    geo_file_path = settings.geofile_path.format(feature_id)
+    qlat_input_folder = settings.qlat_input_path.format(lid)
+    return {
+        "lid": lid,
+        "hy_id": hy_id,
+        "initial_start": initial_start,
+        "start_datetime": start_datetime,
+        "geo_file_path": geo_file_path,
+        "nts": nts,
+        "qlat_input_folder": qlat_input_folder,
+    }
+
+
+def create_initial_start_file(params: Dict[str, str], settings: Settings) -> Path:
+    """Creating the initial start/restart files
+
+    Parmeters:
+    ----------
+    params: Dict[str, str]
+        The parameters from the API to be added to the t-route config file
+    settings: Settings
+        The T-route BaseSettings
+
+    Returns:
+    --------
+    Path:
+        The path to the t-route restart file
+    """
+    start_datetime = datetime.strptime(params["start_datetime"], "%Y-%m-%d_%H:%M")
+    formatted_datetime = start_datetime.strftime("%Y-%m-%d_%H:%M")
+
+    gdf = gpd.read_file(params["geo_file_path"], layer="network")
+    mask = gdf["divide_id"].isna()
+    keys = [int(val[4:]) for val in set(gdf[~mask]["divide_id"].values.tolist())]
```

Comment:
I strongly encourage using string split semantics instead of assuming prefix lengths are always going to be the same.